src/execution/function_ref.rs

```diff
@@ -0,0 +1,35 @@
+use alloc::string::{String, ToString};
+
+use crate::execution::{hooks::HookSet, value::InteropValueList, RuntimeInstance};
+use crate::RuntimeError;
+
+pub struct FunctionRef {
+    pub(crate) module_name: String,
+    pub(crate) function_name: String,
+    pub(crate) module_index: usize,
+    pub(crate) function_index: usize,
+}
+
+impl FunctionRef {
+    pub(crate) fn new(
+        module_name: &str,
+        function_name: &str,
+        module_index: usize,
+        function_index: usize,
+    ) -> Self {
+        Self {
+            module_name: module_name.to_string(),
+            function_name: function_name.to_string(),
+            module_index,
+            function_index,
+        }
+    }
+
+    pub fn invoke<H: HookSet, Param: InteropValueList, Returns: InteropValueList>(
+        &self,
+        runtime: &mut RuntimeInstance<H>,
+        params: Param,
+    ) -> Result<Returns, RuntimeError> {
+        runtime.invoke(self, params)
+    }

```

Comment:
I don't think we need this method. The user should call `runtime.invoke(...)` directly (the tests already do it like this).

Reply:
I'm indifferent. I added it to see what other people think about it. We'll discuss this at the meeting and go on from there.